application/src/main/java/org/togetherjava/tjbot/commands/moderation/WarnCommand.java

```diff
@@ -0,0 +1,165 @@
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.utils.Result;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.WarnSystem;
+import org.togetherjava.tjbot.db.generated.tables.records.WarnSystemRecord;
+
+import java.awt.*;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+
+/**
+ * This command allows a moderator to warn a user. This moderator will need to have ban permission
+ * and well need to type {@code /warn @user reason}. If the command is successful the user will be
+ * warned and will receive a dm telling them why there were warned. After that all the data will be
+ * added to the database.
+ */
+public class WarnCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(WarnCommand.class);
+    private static final String USER_OPTION = "user";
+    private static final String REASON_OPTION = "reason";
+    private static final String ACTION_VERB = "warn";
+    private final Database database;
+    private final Predicate<String> hasRequiredRole;
+
+    /**
+     * Creates a new Instance.
+     */
+    public WarnCommand(@NotNull Database database) {
+        super("warn", "warns the user", SlashCommandVisibility.GUILD);
+        this.database = database;
+
+        getData().addOption(OptionType.USER, USER_OPTION, "The user to warn", true)
+            .addOption(OptionType.STRING, REASON_OPTION, "The reason for the warning", true);
+
+        hasRequiredRole = Pattern.compile(Config.getInstance().getSoftModerationRolePattern())
+            .asMatchPredicate();
+    }
+
+    private static @NotNull RestAction<Boolean> dmUser(long userId, @NotNull String reason,
+            @NotNull Guild guild, @NotNull SlashCommandEvent event) {
+        return event.getJDA()
+            .openPrivateChannelById(userId)
+            .flatMap(privateChannel -> event.replyEmbeds(new EmbedBuilder().setTitle("Warned")
+                .setDescription("You have been warned in " + guild.getName() + " for " + reason)
+                .setColor(Color.decode("#895FE8"))
+                .build()))
```

Comment:
Follow the existing style and dont use embeds. Otherwise the UX will be very confusing.